flow/types.js

```diff
@@ -86,27 +90,33 @@ declare module 'icarus-backend' {
     data: any
   }
 
-  declare type TxInput = {
-    txDbId: number,
+  declare type TxInputDbResult = {
+    txDbId: string,
     address: string,
     value: number,
     hash: string,
     index: number,
     isGenesis?: boolean,
+    txOutDbId: string,
   }
 
- declare type TxOutput = {
-    txDbId: number,
+ declare type TxOutputDbResult = {
+    txDbId: string,
     address: string,
     value: number,
     index: number,
+    txOutDbId: string,
+  }
+
+  declare type WithTokens<T> = T & {
```

Comment:
unless there is a very good reason I would not deal with outputs with tokens and without tokens as it brings low-level concerns (joining utxos with tokens) outside of the data fetching layer which can become a madness pretty quickly if it already didn't